CPP_STYLE_GUIDE.md

```diff
@@ -316,13 +315,23 @@ exports.foo = function(str) {
 
 #### Avoid throwing JavaScript errors in nested C++ methods
 
-When you have to throw the errors from C++, try to do it at the top level and
-not inside of nested calls.
+When you need to throw a JavaScript exception from C++ (i.e.
+`isolate()->ThrowException()`) prefer to do it as close to the return to JS as
+possible, and not inside of nested C++ calls. Since this changes the `isolate`'s
+state doing closest to where it is consumed, reduces the chance of side effects.
 
-Using C++ `throw` is not allowed.
+Node.js is built without C++ exception semantics, so code using `throw` or, even
```

Comment:
s/exception semantics/exceptions/?

Reply:
I went with `exception handling` and added a reference (GCC state 7% runtime cost which I think is interesting).